multicluster/cmd/multicluster-controller/clusterset_webhook.go

```diff
@@ -57,6 +63,20 @@ func (v *clusterSetValidator) Handle(ctx context.Context, req admission.Request)
 			klog.ErrorS(err, "the field 'clusterID' is immutable", "ClusterSet", klog.KObj(clusterSet))
 			return admission.Denied("the field 'clusterID' is immutable")
 		}
+
+		ui := req.UserInfo
+		_, saName, err := serviceaccount.SplitUsername(ui.Username)
+		if err != nil {
+			klog.ErrorS(err, "Error getting ServiceAccount name", "ClusterSet", req.Namespace+"/"+req.Name)
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+		if saName == mcControllerSAName {
```

Comment:
This block is useless and can be removed.

Reply:
You are right, removed it.